packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -53,6 +66,265 @@ This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aw
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+  userPools: [userpool],
+  defaultClientOptions: {
+    userPoolClientName: 'my-user-pool-client',
+    disableOath: true,
+    refreshTokenValidity: Duration.days(14),
+  },
+});
+```
+
+You can also associate user pools with an identity pool after instantiation. The Identity Pool's `addUserPool` method returns the User Pool Client that has been created:
+
+```ts
+const userPoolClient = identityPool.addUserPool(userpool, {
+  userPoolClientName: 'my-user-pool-client',
+  disableOath: true,
+  refreshTokenValidity: Duration.days(14),
+});
+```
+
+#### Associating a Provider Directly
+
+You can associate with one or more [external identity providers](https://docs.aws.amazon.com/cognito/latest/developerguide/external-identity-providers.html) directly with an identity pool using `IdentityPoolProps.supportedLoginProviders`:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+  supportedLoginProviders: {
+    amazon: 'amzn1.application.12312k3j234j13rjiwuenf',
+    facebook: '1234567890123',
+    google: '12345678012.apps.googleusercontent.com',
+    apple: 'com.yourappleapp.auth',
+    twitter: 'my-twitter-id;my-twitter-secret',
+  },
+});
+```
+
+If you want to associate more than one provider of the same type with the identity pool, you can do so using User Pools, OpenIdConnect, or Saml. You can't attach more than one provider per external service directly to the identity pool.
+
+
+### OpenId Connect and Saml
+
+[OpenID Connect](https://docs.aws.amazon.com/cognito/latest/developerguide/open-id.html) is an open standard for authentication that is supported by a number of login providers. Amazon Cognito supports linking of identities with OpenID Connect providers that are configured through [AWS Identity and Access Management](http://aws.amazon.com/iam/). 
+
+You can use an identity provider that supports [Security Assertion Markup Language 2.0 (SAML 2.0)](https://docs.aws.amazon.com/cognito/latest/developerguide/saml-identity-provider.html) to provide a simple onboarding flow for your users. Your SAML-supporting identity provider specifies the IAM roles that can be assumed by your users so that different users can be granted different sets of permissions.
+Associating an OpenId Connect or Saml provider with an identity pool:
+
+```ts
+const openIdConnectProvider = iam.OpenIdConnectProvider.fromOpenIdConnectProviderArn(this, 'my-openid-connect-provider', 'arn:aws:iam::123456789012:oidc-provider/my-open-id-provider.example.com');
+
+const samlProvider = iam.SamlProvider.fromSamlProviderArn(this, 'my-saml-provider', 'arn:aws:iam::123456789012:saml-provider/my-saml-provider');
```

Comment:
Examples should specify how to use the provider natively in the CDK. This should not be based on the import APIs.

```suggestion
const openIdConnectProvider = new iam.OpenIdConnectProvider(this, 'my-openid-connect-provider', ...);

const samlProvider = new iam.SamlProvider(this, 'my-saml-provider', ...);
```